components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1501,7 +1513,11 @@ private String handleOAuthAuthorizationRequest(OAuthMessage oAuthMessage)
             setSPAttributeToRequest(oAuthMessage.getRequest(), validationResponse.getApplicationName(), tenantDomain);
         }
 
-        OAuthAuthzRequest oauthRequest = new CarbonOAuthAuthzRequest(oAuthMessage.getRequest());
+        String oauthAuthzRequestClassName = OAuthServerConfiguration.getInstance().getOAuthAuthzRequestClass();
+
+        OAuthAuthzRequest oauthRequest = defaultOAuthAuthzRequestClassName.equals(oauthAuthzRequestClassName) ?
+                new CarbonOAuthAuthzRequest(oAuthMessage.getRequest()) :
+                getOAuthAuthzRequest(oauthAuthzRequestClassName, oAuthMessage.getRequest());
```

Comment:
Lets move the logic of checking if the authz request class is default to a separate method and use it inside the getOAuthAuthzRequest method.

Reply:
The class resolving logic can also be in the `getOAuthAuthzRequest` method

Reply:
Fixed with 79bab73a362a2021f071c540d7aed7a385ff3817